### PR TITLE
Add stack using linked list

### PR DESCRIPTION
**Pull request by muthuannamalai12**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
Added Stack Implementation using Linked List in C.
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1983
(Write your answer here.)

### CODE REVIEW

**Comment by muthuannamalai12** [comment]
@atarax665 Kindly Review

File: C/README.md [hunk 1]
```diff
@@ -58,7 +58,7 @@
 - [Bellman Ford Algorithm](ds/bellman_ford.c)
 - [Implementation of Dijkstra's Algorithm](ds/Dijsktra.c)
 - [Circular Queue](ds/circularqueue.c)
-- [Post Order Traversal](ds/Postorder.c)
```
Comment (atarax665) [inline C/README.md:61]: Don't remove anything from the README. You just need to add the link of your program under appropriate section.

File: C/ds/StackImplementationWithLinkedList.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+Enter your choice 
+1 Enter node data 25
+
+*****MENU****** 
+1.Push
+2.Pop 
+3.Display 
+4.Exit
+Enter your choice 3 
+data is 23 25
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]:
```suggestion
Data is 23 25
```

File: C/ds/StackImplementationWithLinkedList.c [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		{
+			printf("%d\n", temp->data);
+			temp = temp->link;
+		}
+	}
+}
+void main()
+	{
+		int choice;
+		do {
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]: Add ```printf("\n*****MENU******\n");``` to have the output similar to the Sample I/O.

File: C/ds/StackImplementationWithLinkedList.c [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	}
+}
+void display()
+{
+	struct node *temp;
+	if (top == NULL)
+		printf("\n Stack is empty");
+	else
+	{
+		temp = top;
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]: Add a line ```printf("\nData is "); before the while loop.

File: C/ds/StackImplementationWithLinkedList.c [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+		while (temp != NULL)
+		{
+			printf("%d\n", temp->data);
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]:
```suggestion
			printf("%d ", temp->data);
```

File: C/ds/StackImplementationWithLinkedList.c [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+2.Pop 
+3.Display 
+4.Exit
+Enter your choice 3 
+data is 23 25
+
+Time complexity : O(1)
+Space Complexity: O(n)
+*/
+
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]: Remove this extra blank line.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by muthuannamalai12** [comment]
I have changed it @atarax665 Kindly review it again

File: C/ds/StackImplementationWithLinkedList.c [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+					break;
+				case 4:
+					exit(0);
+				default:
+					printf("Invalid choice");
+					}
+}while(choice < 5); 
+getchar();
+}
+
```
Comment (HarshCasper) [inline C/ds/StackImplementationWithLinkedList.c]: Inconsistent indentation. Please fix it 👍

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Code looks good. Tiny change required and ready to merge 👍

**Comment by muthuannamalai12** [comment]
I have changed it. Kindly review again @HarshCasper

File: C/ds/StackImplementationWithLinkedList.c [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        while (temp != NULL) {
+            printf("%d ", temp -> data);
+            temp = temp -> link;
+        }
+    }
+}
+void main() {
+        int choice;
+        do {
+            printf("\n*****Menu*****)
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]:
```suggestion
            printf("\n*****Menu*****);
```

File: C/ds/StackImplementationWithLinkedList.c [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+                printf("\n1.Push\n2.Pop\n3.Display\n4.Exit\n"); printf("Enter your choice:"); scanf("%d", & choice);
```
Comment (atarax665) [inline C/ds/StackImplementationWithLinkedList.c]: Indentation is still inconsistent.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by muthuannamalai12** [comment]
I have changed it. Kindly review again @atarax665

File: C/README.md (hunk 1 above)
Comment (atarax665) [reply C/README.md:61]: This issue has not been resolved yet.

**Comment by muthuannamalai12** [comment]
I have resolved the change in README.md. Kindly review again @atarax665

**Review by vatsalkesarwani12 — APPROVED** [review]
update readme rest LGTM

**Comment by muthuannamalai12** [comment]
@atarax665 @HarshCasper @iamrajiv  Kindly Review

File: C/README.md [hunk 10]
```diff
@@ -58,8 +58,8 @@
...
-- [Stack Implementation with Array](ds/StackImplementationWithArray.c)
```
Comment (vatsalkesarwani12) [inline C/README.md:62]: This is missing

**Comment by muthuannamalai12** [comment]
@plazzy99 I have resolved it Kindly review again

**Comment by muthuannamalai12** [comment]
@plazzy99 I have updated it. Kindly review again

**Comment by muthuannamalai12** [comment]
@atarax665 @HarshCasper @iamrajiv Kindly Review

**Comment by muthuannamalai12** [comment]
@atarax665 @HarshCasper @iamrajiv @plazzy99. Kindly Review

**Review by atarax665 — APPROVED** [review]

**Comment by muthuannamalai12** [comment]
@HarshCasper @atarax665. Can you please add DWOC and level label for this issue here on my PR.

**Comment by muthuannamalai12** [comment]
@HarshCasper @atarax665 @anushkrishnav. Can you please add DWOC and level label for this issue here on my PR.

**Comment by muthuannamalai12** [comment]
@HarshCasper @atarax665 @anushkrishnav . Can you please add DWOC and level label for this issue here on my PR.

**Comment by muthuannamalai12** [comment]
@HarshCasper @atarax6 @anushkrishnav. Can you please add a level label for this issue here on my PR.

**Pull request merged**